### PR TITLE
ncm-metaconfig: Add support for keepalived

### PR DESCRIPTION
**Pull request by apdibbo**
Adds support for Keepalived to provide floating IP Addresses for use in Load Balancers

### CODE REVIEW

File: ncm-metaconfig/src/main/metaconfig/keepalived/global.tt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[% FOREACH item IN global_defs.pairs -%]
+[%     item.key %] [% item.value -%]
```
Comment (stdweird) [inline ncm-metaconfig/src/main/metaconfig/keepalived/global.tt]: the ending `-%]` will put multiple global_defs on the same line, make sure your test example has more than on entry

File: ncm-metaconfig/src/main/metaconfig/keepalived/main.tt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+global_defs {
+[% INCLUDE 'metaconfig/keepalived/global.tt' FILTER indent -%]
+}
+[% FOREACH vrrp_script IN vrrp_scripts -%]
+vrrp_script [% vrrp_script.name %] {
+[% INCLUDE 'metaconfig/keepalived/vrrp_script.tt' FILTER indent -%]
```
Comment (stdweird) [inline ncm-metaconfig/src/main/metaconfig/keepalived/main.tt]: indent in TT code

File: ncm-metaconfig/src/main/metaconfig/keepalived/main.tt [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+[% INCLUDE 'metaconfig/keepalived/global.tt' FILTER indent -%]
+}
+[% FOREACH vrrp_script IN vrrp_scripts -%]
+vrrp_script [% vrrp_script.name %] {
+[% INCLUDE 'metaconfig/keepalived/vrrp_script.tt' FILTER indent -%] 
+}
+[% END %]
+[% FOREACH vrrp_instance IN vrrp_instances -%]
+vrrp_instance [% vrrp_instance.name %] {
+[%- INCLUDE 'metaconfig/keepalived/vrrp_instance.tt' FILTER indent -%]
```
Comment (stdweird) [inline ncm-metaconfig/src/main/metaconfig/keepalived/main.tt]: indet in TT code

File: ncm-metaconfig/src/main/metaconfig/keepalived/virtual_ipaddress.tt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+[% FOREACH vip IN vrrp_instance.virtual_ipaddresses -%]
+[% vip.ipaddress %] dev [% vip.interface %]
```
Comment (stdweird) [inline ncm-metaconfig/src/main/metaconfig/keepalived/virtual_ipaddress.tt]: indent in TT code

**Comment by jrha** [comment]
@alahiff?

**Pull request merged**